### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by Iguins**
GitHub Issue: #

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [x] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+`await _biometryService.Encrypt(ct, "Key", "StringToEnrypt");`
+
+- IOS
+
+The `SecKeyChain` will be use to store a string linked to a key. IOS is in charge to secure the data with biometric Authentication during the process.
+In case of error, it throw a `SecurityException`.
+
+- Android
+
+A new `CryptoObject` from `AndroidX.Biometric` is created with a key as parameter. Then the data will be encrypted and presented to the `biometricPrompt` manager.
```
Comment (DanielCauser) [inline README.md:116]: >(...)a key as `a` parameter. Then the(...)

File: README.md [hunk 2]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+The final step will encode the data in base64 and store it in App with the shared preferences.
+
+
 ### Decrypt
 
+The follow method do specific actions according to the platorm targeted.
```
Comment (DanielCauser) [inline README.md:122]: >The `following` method `does` specific actions according to the `platform` targeted.

Typo

File: README.md [hunk 3]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+`await _biometryService.Decrypt(ct, "Secret");`
+
+- IOS
+
+Retrieve the encrypted data from the `SecKeyChain` with the secret as parameter. IOS is in charge to decrypt the data with biometric Authentication during the process.
```
Comment (DanielCauser) [inline README.md:127]: >(...) with the secret as `a` parameter. IOS is in charge `of decrypting` the data(...)

typo

File: README.md [hunk 4]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+The follow method do specific actions according to the platorm targeted.
+`await _biometryService.Decrypt(ct, "Secret");`
+
+- IOS
+
+Retrieve the encrypted data from the `SecKeyChain` with the secret as parameter. IOS is in charge to decrypt the data with biometric Authentication during the process. 
+
+- Android
+
+Retrieve the shared preference encrypted data, then decrypt it with the secret as parameter by presenting it to the `biometricPrompt` manager.
```
Comment (DanielCauser) [inline README.md:131]: >(...)the secret as `a` parameter by presenting(...)

typo

**Review by DanielCauser — CHANGES_REQUESTED** [review]
Left some comments regarding some types, as well as rewording some sentences.

**Review by DanielCauser — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+| Feature          | IOS | Android | UWP  |
+| ------------------ | ----- | --------- | ------ |
+| GetCapability    | x   | x       | x    |
+| ValidateIdentity | x   | x       | Mock |
+| Encrypt          | x   | x       | Mock |
+| Decrypt          | x   | x       | Mock |
+
+### instantiation
+
+- IOS
```
Comment (jeanplevesque) [inline README.md:39]: use level 4 headers instead of bullet points

File: README.md [hunk 6]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+
+
+| Feature          | IOS | Android | UWP  |
+| ------------------ | ----- | --------- | ------ |
+| GetCapability    | x   | x       | x    |
+| ValidateIdentity | x   | x       | Mock |
+| Encrypt          | x   | x       | Mock |
+| Decrypt          | x   | x       | Mock |
+
+### instantiation
```
Comment (jeanplevesque) [inline README.md:37]: titles should start with a capital letter

File: README.md [hunk 7]
```diff
@@ -1,10 +1,12 @@
 # Biometry Service
 
 <!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
+
 [![All Contributors](https://img.shields.io/badge/all_contributors-5-orange.svg?style=flat-square)](#contributors-)
+
 <!-- ALL-CONTRIBUTORS-BADGE:END -->
 
-This library offer a simple contract to use the biometry across Android, IOS and UWP. 
+This library offers a simple contract to use the biometry across Android, IOS and UWP.
```
Comment (jeanplevesque) [inline README.md:9]: it's iOS not IOS (to change everywhere)

File: README.md [hunk 8]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
 - Encryt
 - Decrypt
 
+As of now, this is the list of features available per platform.
+
+
+| Feature          | IOS | Android | UWP  |
+| ------------------ | ----- | --------- | ------ |
```
Comment (jeanplevesque) [inline README.md:31]: you can probably align this

File: README.md [hunk 9]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+| Decrypt          | x   | x       | Mock |
+
+### instantiation
+
+- IOS
+
+An example of instantiation as follow with the fallback to the pin code with some text descriptions to display for the user.
+
+```
+            var options = new BiometryOptions();
```
Comment (jeanplevesque) [inline README.md:44]: do we really need all that indentation?

File: README.md [hunk 10]
```diff
@@ -22,16 +24,112 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+- Android
+
+Face ID is only available with the followwing configuration `.SetAllowedAuthenticators(BiometricManager.Authenticators.BiometricWeak` during instantiation of the service.
+
+Encrypt/Decrypt method are only available with the following configuration `.SetAllowedAuthenticators(BiometricManager.Authenticators.BiometricStrong` during instantiation of the service.
+
+An example of instantiation is as follow, with the fallback to the pin code with some text descriptions to display for the user.
+
+```
+			    _biometryService = new BiometryService(MainActivity.Instance,
```
Comment (jeanplevesque) [inline README.md:63]: same here... seems to be lots of unnecessary indentation

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (Iguins) [reply README.md:44]: Nop, I just messed with my Editor.

File: README.md (hunk 8 above)
Comment (Iguins) [reply README.md:31]: Yeah tried, it's doing weird stuff when I do :(

File: README.md [hunk 11]
```diff
@@ -22,16 +24,115 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+
+| Feature          | iOS | Android | UWP  |
+| ------------------ | ----- | --------- | ------ |
+| GetCapability    | x   | x       | x    |
+| ValidateIdentity | x   | x       | Mock |
+| Encrypt          | x   | x       | Mock |
+| Decrypt          | x   | x       | Mock |
+
+## Instantiation
+
```
Comment (jeanplevesque) [inline README.md:38]: Shouldn't there be more content under the Getting Started header?

File: README.md [hunk 12]
```diff
@@ -22,16 +24,115 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+#### iOS
+
+An example of instantiation as follow with the fallback to the pin code with some text descriptions to display for the user.
+
+```
+var options = new BiometryOptions();
+    options.LocalizedReasonBodyText = "REASON THAT APP WANTS TO USE BIOMETRY :)";
```
Comment (jeanplevesque) [inline README.md:45]: code is not usually indented like this.

File: README.md [hunk 13]
```diff
@@ -22,16 +24,115 @@ The biometryService Interface 'IBiometryService' Implement the following method
...
+    options.LocalizedFallbackButtonText = "FALLBACK";
+    options.LocalizedCancelButtonText = "CANCEL";
+
+// use LAPolicy.DeviceOwnerAuthenticationWithBiometrics for biometrics only with no fallback to passcode/password
+// use LAPolicy.DeviceOwnerAuthentication for biometrics+watch with fallback to passcode/password
+    _biometryService = new BiometryService(options, async ct => "Biometrics_Confirm", LAPolicy.DeviceOwnerAuthentication);
```
Comment (jeanplevesque) [inline README.md:51]: indentation is probably wrong here too

**Review by jeanplevesque — APPROVED** [review]

**Pull request merged**